packages/protocol/contracts/common/MintGoldSchedule.sol

```diff
@@ -0,0 +1,265 @@
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+import "openzeppelin-solidity/contracts/utils/Address.sol";
+
+import "../common/FixidityLib.sol";
+import "../common/libraries/ReentrancyGuard.sol";
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+import "../common/interfaces/IGoldToken.sol";
+import "../../contracts-0.8/common/IsL2Check.sol";
+
+/**
+ * @title Contract for minting new CELO token based on a schedule.
+ */
+contract MintGoldSchedule is UsingRegistry, ReentrancyGuard, Initializable, IsL2Check {
+  using SafeMath for uint256;
+  using FixidityLib for FixidityLib.Fraction;
+  using Address for address payable; // prettier-ignore
+
+  uint256 constant GENESIS_GOLD_SUPPLY = 600000000 ether; // 600 million Gold
+  uint256 constant GOLD_SUPPLY_CAP = 1000000000 ether; // 1 billion Gold
+  uint256 constant YEARS_LINEAR = 15;
+  uint256 constant SECONDS_LINEAR = YEARS_LINEAR * 365 * 1 days;
+
+  uint256 public genesisStartTime = 1587587214; // Copied over from `EpochRewards().startTime()`.
+  uint256 public l2StartTime;
+  uint256 public totalSupplyAtL2Start;
+
+  uint256 public totalMintedBySchedule;
+  address public communityRewardFund;
+  address public carbonOffsettingPartner;
+
+  FixidityLib.Fraction private communityRewardFraction;
+  FixidityLib.Fraction private carbonOffsettingFraction;
+
+  event CommunityRewardFractionSet(uint256 fraction);
+  event CarbonOffsettingFundSet(address indexed partner, uint256 fraction);
+
+  /**
+   * @notice Sets initialized == true on implementation contracts
+   * @param test Set to true to skip implementation initialization
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @notice A constructor for initialising a new instance of a MintGoldSchedule contract.
+   * @param _l2StartTime timestamp for L1 to L2 transition
+   * @param _communityRewardFraction The percentage of rewards that go the community funds.
+   * @param _carbonOffsettingPartner The address of the carbon offsetting partner.
+   * @param _carbonOffsettingFraction The percentage of rewards going to carbon offsetting partner.
+   * @param registryAddress Address of the deployed contracts registry.
+   */
+  function initialize(
+    uint256 _l2StartTime,
+    uint256 _communityRewardFraction,
+    address _carbonOffsettingPartner,
+    uint256 _carbonOffsettingFraction,
+    address registryAddress
+  ) external initializer {
+    _transferOwnership(msg.sender);
+    require(registryAddress != address(0), "The registry address cannot be the zero address");
+    setRegistry(registryAddress);
+    communityRewardFund = address(getGovernance());
+    setCommunityRewardFraction(_communityRewardFraction);
+    setCarbonOffsettingFund(_carbonOffsettingPartner, _carbonOffsettingFraction);
+    totalSupplyAtL2Start = getGoldToken().totalSupply();
```

Comment:
Since the release of core contracts will most probably happen still during L1 we need to be able to set this variable after the contract is initialised.